buffer/src/main/java/io/netty/buffer/api/internal/Statics.java

```diff
@@ -230,11 +232,22 @@ static void unsafeSetDrop(ResourceSupport<?, ?> obj, Drop<?> replacement) {
     }
 
     static CharSequence readCharSequence(Buffer source, int length, Charset charset) {
+        try {
+            return copyToCharSequence(source, source.readerOffset(), length, charset);
+        } finally {
+            source.skipReadable(length);
```

Comment:
Should we really update the index if we throw ?

Reply:
hmm good point